visualization/app/codeCharta/state/effects/nodeContextMenu/nodeContextMenu.service.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { ApplicationInitStatus } from "@angular/core"
+import { TestBed } from "@angular/core/testing"
+import { MaterialModule } from "../../../../material/material.module"
+import { EffectsModule } from "../../angular-redux/effects/effects.module"
+import { NodeContextMenuService } from "./nodeContextMenu.service"
+import { NodeContextMenuCardModule } from "./nodeContextMenuCard/nodeContextMenuCard.module"
+import { OpenNodeContextMenuEffect } from "./openNodeContextMenu.effect"
+
+describe("nodeContextMenuService", () => {
+	let mockedWheelTargetElement
+
+	beforeEach(async () => {
+		mockedWheelTargetElement = { addEventListener: jest.fn(), removeEventListener: jest.fn() }
+		// @ts-ignore
+		jest.spyOn(document, "getElementById").mockImplementation(() => mockedWheelTargetElement)
+
+		TestBed.configureTestingModule({
+			imports: [MaterialModule, EffectsModule.forRoot([OpenNodeContextMenuEffect]), NodeContextMenuCardModule],
+			providers: [NodeContextMenuService]
+		})
+		await TestBed.inject(ApplicationInitStatus).donePromise
+	})
+
+	it("should close on document click", () => {
+		const nodeContextMenu = TestBed.inject(NodeContextMenuService)
+		const closeSpy = jest.spyOn(nodeContextMenu, "close")
+		nodeContextMenu.open(10, 10)
+
+		document.dispatchEvent(new MouseEvent("click"))
+
+		expect(closeSpy).toHaveBeenCalled()
+	})
+
+	it("should not close when clicking within (its) color picker", () => {
```

Comment:
When I tested this test case in the application, the menu still closed on almost every mouse click. In our demo version this does not happen. Could you maybe take a look at it again?

Reply:
As far as I know the following is the only case, where the ctx menu should not close. And this works on my machine :S
![image](https://user-images.githubusercontent.com/3596742/152021268-8be586e4-8dbc-4654-b43b-99449ac48d8d.png)